tests/lib/rules/camelcase.js

```diff
@@ -407,6 +407,35 @@ ruleTester.run("camelcase", rule, {
             code: "class C { snake_case; #snake_case; #snake_case2() {} }",
             options: [{ properties: "never" }],
             parserOptions: { ecmaVersion: 2022 }
+        },
+
+        // Combinations of `properties` and `ignoreDestructring`
+        {
+            code: `
+            const { some_property } = obj;
+
+            const foo = ({ some_property }) => {
+                console.log(some_property)
+            };
+
+            function bar({ some_property }) {
+                console.log(some_property)
+            };
+            `,
+            options: [{ properties: "never", ignoreDestructuring: true }],
+            parserOptions: { ecmaVersion: 2022 },
+            errors: [{ messageId: "notCamelCase", data: { name: "some_property" } }]
+        },
```

Comment:
This test case might be confusing, as it doesn't have identifiers where `properties: "never"` applies. The same code is valid with the default `properties: "always"`, too - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhbWVsY2FzZTogW1wiZXJyb3JcIiwgeyBwcm9wZXJ0aWVzOiBcImFsd2F5c1wiLCBpZ25vcmVEZXN0cnVjdHVyaW5nOiB0cnVlIH1dICovXG5cbmNvbnN0IHsgc29tZV9wcm9wZXJ0eSB9ID0gb2JqO1xuXG5jb25zdCBmb28gPSAoeyBzb21lX3Byb3BlcnR5IH0pID0+IHtcbiAgICBjb25zb2xlLmxvZyhzb21lX3Byb3BlcnR5KVxufTtcblxuZnVuY3Rpb24gYmFyKHsgc29tZV9wcm9wZXJ0eSB9KSB7XG4gICAgY29uc29sZS5sb2coc29tZV9wcm9wZXJ0eSlcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

`properties: "never"` applies only to identifiers used to create properties. Identifiers used to read properties are always allowed.

```js
/* eslint camelcase: ["error", { properties: "always", ignoreDestructuring: true }] */

x = obj.not_camelcase; // always allowed

obj.not_camelcase = x; // disallowed. can be allowed with properties: "never"

({ not_camelcase: x } = obj ); // always allowed

obj = { not_camelcase: x }; // // disallowed. can be allowed with properties: "never"


```